src/python/pants/backend/python/lint/docformatter/rules.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from pants.backend.python.lint.docformatter.subsystem import Docformatter
+from pants.backend.python.lint.python_format_target import PythonFormatTarget
+from pants.backend.python.lint.python_lint_target import PythonLintTarget
+from pants.backend.python.rules import download_pex_bin, pex
+from pants.backend.python.rules.pex import (
+  CreatePex,
+  Pex,
+  PexInterpreterConstraints,
+  PexRequirements,
+)
+from pants.backend.python.subsystems import python_native_code, subprocess_environment
+from pants.backend.python.subsystems.subprocess_environment import SubprocessEncodingEnvironment
+from pants.engine.fs import Digest, DirectoriesToMerge
+from pants.engine.isolated_process import (
+  ExecuteProcessRequest,
+  ExecuteProcessResult,
+  FallibleExecuteProcessResult,
+)
+from pants.engine.legacy.structs import TargetAdaptorWithOrigin
+from pants.engine.rules import UnionRule, rule, subsystem_rule
+from pants.engine.selectors import Get
+from pants.python.python_setup import PythonSetup
+from pants.rules.core import find_target_source_files, strip_source_roots
+from pants.rules.core.find_target_source_files import (
+  FindTargetSourceFilesRequest,
+  TargetSourceFiles,
+)
+from pants.rules.core.fmt import FmtResult
+from pants.rules.core.lint import LintResult
+
+
+@dataclass(frozen=True)
+class DocformatterTarget:
+  adaptor_with_origin: TargetAdaptorWithOrigin
+  prior_formatter_result_digest: Optional[Digest] = None  # unused by `lint`
+
+
+@dataclass(frozen=True)
+class SetupRequest:
+  target: DocformatterTarget
+  check_only: bool
+
+
+@dataclass(frozen=True)
+class Setup:
+  process_request: ExecuteProcessRequest
+
+
+def generate_args(
+  *, source_files: TargetSourceFiles, docformatter: Docformatter, check_only: bool,
+) -> Tuple[str, ...]:
+  return (
+    "--check" if check_only else "--in-place",
+    *docformatter.options.args,
+    *sorted(source_files.snapshot.files),
+  )
+
+
+@rule
+async def setup(
+  request: SetupRequest,
+  docformatter: Docformatter,
+  python_setup: PythonSetup,
+  subprocess_encoding_environment: SubprocessEncodingEnvironment,
+) -> Setup:
+  adaptor_with_origin = request.target.adaptor_with_origin
+  adaptor = adaptor_with_origin.adaptor
+
+  requirements_pex = await Get[Pex](
```

Comment:
Idle thought, definitely out of scope of this PR: This seems to mention the `Docformatter` class a lot... I wonder if we could make `CreatePex` smarter so that I can do `await Get[Pex](docformatter.default_pex)`, and all these fields will be filled out with good defaults for me.

Reply:
Interesting. I definitely think there are grounds for deduplication between the linters and we are starting to have enough stability to justify it, now that we have 6 Python linters and all the functionality is there like precise file arguments.

But before we generalize, we're drafting a proposal to improve the performance of V2 linters and it would entail a lot of churn in these linters. I want to wait to generalize anything until that is all over.

Reply:
Sounds good!